### PR TITLE
Debugger: replace sprintf() part 3 - Debugger_Symbols

### PR DESCRIPTION
**Pull request by kiyolee**
- Change `_CmdSymbolsInfoHeader()` to return `std::string`

### CODE REVIEW

File: source/Debugger/Debugger_Symbols.cpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
 // Utils _ ________________________________________________________________________________________
 
-	void      _CmdSymbolsInfoHeader( int iTable, char * pText, int nDisplaySize = 0 );
+	std::string _CmdSymbolsInfoHeader( int iTable, int nDisplaySize = 0 );
```
Comment (Michaelangel007) [inline source/Debugger/Debugger_Symbols.cpp:78]: Please re-align these 2 other util functions to the top util functions.
i.e. 
```
        std::string _CmdSymbolsInfoHeader( int iTable, int nDisplaySize = 0 );
	void        _PrintCurrentPath();
	Update_t    _PrintSymbolInvalidTable();
```

File: source/Debugger/Debugger_Symbols.cpp [hunk 2]
```diff
@@ -90,33 +90,28 @@ void _PrintCurrentPath()
...
-	char sText[ CONSOLE_WIDTH * 2 ];
-	char sTemp[ CONSOLE_WIDTH * 2 ];
-
 	// TODO: display the user specified file name
 	ConsoleBufferPush( "Invalid symbol table." );
 
-	ConsolePrintFormat( "Only %s%d%s symbol tables are supported:"
-		, CHC_NUM_DEC, NUM_SYMBOL_TABLES
-		, CHC_DEFAULT
+	ConsolePrintFormat( "Only " CHC_NUM_DEC "%d" CHC_DEFAULT " symbol tables are supported:"
```
Comment (Michaelangel007) [inline source/Debugger/Debugger_Symbols.cpp:96]: The "inlining" of the meta-formatting is a tad harder to read now but on the plus side these should be harder to "unbalance", so accepting.

File: source/Debugger/Debugger_Symbols.cpp [hunk 3]
```diff
@@ -330,27 +317,27 @@ Update_t CmdSymbolsInfo (int nArgs)
 		bDisplaySymbolTables = (1 << iWhichTable);
 	}
 
-	//sprintf( sText, "  Symbols  Main: %s%d%s  User: %s%d%s   Source: %s%d%s"
-	// "Main:# Basic:# Asm:# User1:# User2:# Src1:# Src2:# Dos:# Prodos:#
```
Comment (Michaelangel007) [inline source/Debugger/Debugger_Symbols.cpp:334]: This sample output should be kept.

File: source/Debugger/Debugger_Symbols.cpp [hunk 4]
```diff
@@ -330,27 +317,27 @@ Update_t CmdSymbolsInfo (int nArgs)
 		bDisplaySymbolTables = (1 << iWhichTable);
 	}
 
-	//sprintf( sText, "  Symbols  Main: %s%d%s  User: %s%d%s   Source: %s%d%s"
-	// "Main:# Basic:# Asm:# User1:# User2:# Src1:# Src2:# Dos:# Prodos:#
+	std::string const sIndent = "  ";
```
Comment (Michaelangel007) [inline source/Debugger/Debugger_Symbols.cpp:320]: Please re-align the variable names and assignment.
```
	std::string const sIndent = "  ";
	std::string       sText   = sIndent; // Indent new line
```

**Review by Michaelangel007 — CHANGES_REQUESTED** [review]
Couple of minor cleanup needed but looks good overall.

File: source/Debugger/Debugger_Symbols.cpp (hunk 1 above)
Comment (kiyolee) [reply source/Debugger/Debugger_Symbols.cpp:78]: Fixed in new commit.

File: source/Debugger/Debugger_Symbols.cpp (hunk 3 above)
Comment (kiyolee) [reply source/Debugger/Debugger_Symbols.cpp:334]: Restored in new commit.

File: source/Debugger/Debugger_Symbols.cpp (hunk 4 above)
Comment (kiyolee) [reply source/Debugger/Debugger_Symbols.cpp:323]: Fixed in new commit.

**Review by Michaelangel007 — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**